src/GraphQL/Language/AST/NameNode.cs

```diff
@@ -1,21 +1,39 @@
+using System;
+using System.Collections.Generic;
+
 namespace GraphQL.Language.AST
 {
     /// <summary>
     /// Represents a name within a document. This could be the name of a field, type, argument, directive, alias, etc.
     /// </summary>
-    public class NameNode : AbstractNode
+    public readonly struct NameNode : INode
     {
+        /// <summary>
+        /// Initializes a new instance with the specified name.
+        /// </summary>
+        public NameNode(string name, SourceLocation location)
+        {
+            Name = name;
+            SourceLocation = location;
+        }
+
         /// <summary>
         /// Initializes a new instance with the specified name.
         /// </summary>
         public NameNode(string name)
         {
             Name = name;
+            SourceLocation = default;
         }
 
         /// <summary>
         /// Returns the contained name.
         /// </summary>
         public string Name { get; }
+
+        IEnumerable<INode> INode.Children => throw new NotImplementedException();
```

Comment:
```suggestion
        IEnumerable<INode> INode.Children => null;
```

To align behavior with other leaf nodes.

Reply:
It's still safer.